### PR TITLE
osx fixes for go 1.11

### PR DESCRIPTION
**Pull request by kolyshkin**
This is a 3-in-1 story. Please have a seat.

### Part 1.

Go 1.11 release notes say:

> Go 1.11 now requires <...> macOS 10.10 Yosemite or later

So, change the `OSX_VERSION_MIN` in `osxcross` to `10.10`.

### Part 2.

MACOSX SDK tarball that was used before comes without libc++
headers, which are now required by at least crypto/x509 pkg
when compiling for Darwin. These include files are part of xcode,
which can only be downloaded by someone having Apple Developer ID,
which I don't have so this path is not available for me.

An alternative is to reuse the SDK tarball from the
https://github.com/multiarch/crossbuild project. Using it
fixes the following error:

```
# crypto/x509
osxcross: error: cannot find libc++ headers
```

### Part 3.

The new tarball, although, misses the libtool package (apparently
somehow injected into the tarball that was used earlier) needed to
compile miekg/pkcs11 used by cli, which leads to errors like:

```
# github.com/docker/cli/vendor/github.com/miekg/pkcs11
vendor/github.com/miekg/pkcs11/pkcs11.go:28:10: fatal error: 'ltdl.h' file not found
#include <ltdl.h>
          ^
1 error generated.
```

To fix this one, libtool for osx need to be installed. Building it from
source (with osxcross) requires some other dependencies. Using brew is not
possible since it is Linux. Solution: download a binary package from brew
repo, cherry pick the needed stuff (lib and include).

Tested to fix docker-cli compilation issues on OSX using Go 1.11.

### CODE REVIEW

**Comment by kolyshkin** [comment]
@vdemeester @thaJeztah  PTAL

**Comment by thaJeztah** [comment]
> These include files are part of xcode, which can only be downloaded by someone having Apple Developer ID, which I don't have so this path is not available for me.

@kolyshkin I do have a Mac developer subscription; would it be cleaner to update the package we have at dockerproject.org?

File: osx-cross.sh [hunk 1]
```diff
@@ -7,23 +7,38 @@
...
 time git clone https://github.com/tpoechtrager/osxcross.git $OSXCROSS_PATH
 cd $OSXCROSS_PATH
 git checkout -q $OSX_CROSS_COMMIT
 
 echo "Downloading OSX SDK"
-time curl -sSL https://s3.dockerproject.org/darwin/v2/${OSX_SDK}.tar.xz \
-    -o "${OSXCROSS_PATH}/tarballs/${OSX_SDK}.tar.xz"
+time curl -sSL \
+	https://www.dropbox.com/s/yfbesd249w10lpc/${OSX_SDK}.tar.xz \
+	-o "${OSXCROSS_PATH}/tarballs/${OSX_SDK}.tar.xz"
```
Comment (thaJeztah) [inline osx-cross.sh]: Wondering if we need a checksum for this download

**Comment by kolyshkin** [comment]
> I do have a Mac developer subscription; would it be cleaner to update the package we have at dockerproject.org?

I think we can reuse the one from this PR, it works fine and needs nothing.

It would be awesome though if you can upload it to s3.dockerproject.org. I tried to do it myself, logged in to aws successfully (using kir / docker-core credentials) but accessing S3 gives me "access denied".

File: osx-cross.sh (hunk 1 above)
Comment (kolyshkin) [reply osx-cross.sh]: makes sense

**Comment by andrewhsu** [comment]
```
$ curl -Is https://s3.dockerproject.org/darwin/v2/MacOSX10.11.sdk.tar.xz | head -1
HTTP/2 200
```

**Comment by andrewhsu** [comment]
```
$ curl -Is https://s3.dockerproject.org/darwin/v2/MacOSX10.10.sdk.tar.xz | head -1
HTTP/2 200
```

File: Dockerfile [hunk 2]
```diff
@@ -1,10 +1,10 @@
-FROM    golang:1.11.0
+FROM    golang:1.11.1
```
Comment (thaJeztah) [inline Dockerfile]: Just thinking of this; perhaps good to split the bump to a separate commit, so that we could build an updated / fixed 1.11.0 image as well if we see a need for it

Comment (kolyshkin) [reply Dockerfile]: Sure can

Comment (kolyshkin) [reply Dockerfile]: done
